code/src/hooks/use-sign-in/useSignIn.ts

```diff
@@ -0,0 +1,30 @@
+import { useSignInMutation } from "@/store/api/authApi";
+import { authenticate } from "@/store/slices/userSlice";
+import { SignInCredentials } from "@/types/auth.types";
+import useSnackbar from "@/hooks/use-snackbar/useSnackbar";
+import { useAppDispatch } from "@/hooks/use-redux/useRedux";
+
+const useSignIn = () => {
+  const [signIn, options] = useSignInMutation();
+  const { openSnackbarWithTimeout } = useSnackbar();
+  const dispatch = useAppDispatch();
+
+  const handleSignIn = async (credentials: SignInCredentials) => {
+    const { data: userDetails, error } = await signIn(credentials);
```

Comment:
maybe also should handle the loading here ? What do you think ?

Reply:
I don't think it is good idea. We can use isLoading externally and attach it to some ui as it is present on `options` object